### PR TITLE
Add a couple IPv6 tests

### PR DESCRIPTION
**Pull request by 1st1**
I'm adding a few IPv6 tests. Please review and feel free to suggest adding more.

@fantix need to make sure our CLI passes them.
@jaclarke please update edgedb-js
@fmoor please update edgedb-go

### CODE REVIEW

File: connection_testcases.json [hunk 1]
```diff
@@ -2255,5 +2255,63 @@
...
+      "password": "123123",
+      "serverSettings": {},
+      "tlsCAData": null,
+      "tlsSecurity": "strict",
+      "user": "user3"
+    }
+  },
+  {
+    "opts": {
+      "dsn": "edgedb://user3@[fe80::1ff:fe23:4567:890a]:3000/ab"
```
Comment (elprans) [inline connection_testcases.json:2275]: I'd also add a scoped address test, eg. `[fe80::1ff:fe23:4567:890a%25eth0]`

Comment (1st1) [reply connection_testcases.json]: done

File: connection_testcases.json [hunk 2]
```diff
@@ -2255,5 +2255,63 @@
...
+      "tlsSecurity": "strict",
+      "user": "user3"
+    }
+  },
+  {
+    "opts": {
+      "dsn": "edgedb://user3@[fe80::1ff:fe23:4567:890a%25eth0]:3000/ab"
+    },
+    "result": {
+      "address": ["fe80::1ff:fe23:4567:890a%25eth0", 3000],
```
Comment (elprans) [inline connection_testcases.json:2278]: This should be `fe80::1ff:fe23:4567:890a%eth0`, you need to `urllib.parse.unquote` it

Comment (1st1) [reply connection_testcases.json:2278]: oooh good catch!

Comment (1st1) [reply connection_testcases.json]: I've updated the tests -- please review again

**Review by elprans — APPROVED** [review]

**Comment by fantix** [comment]
@1st1 The CLI can't support scoped address at this moment because of this upstream issue: servo/rust-url#424. This is also an issue in Rust: rust-lang/rust#65976, so even if we parse the URL manually, we're likely unable to use the scoping. cc @tailhook 

I'll fix the remaining IPv6 issue in `edgedb-rust`.

**Review by fmoor — APPROVED** [review]

**Comment by 1st1** [comment]
@fantix 

> the CLI can't support scoped address

Can you make sure we support and parse regular unscoped IPv6 addresses? Scoped ones are incredibly rarely used anyways.

**Comment by fantix** [comment]
> Can you make sure we support and parse regular unscoped IPv6 addresses?

Sure thing.

> we're likely unable to use the scoping

Oh actually Rust would call system API to parse the scoped IPv6 addresses if the Rust-version `Ipv6Addr` fails to parse them. And the parse result is properly scoped, so we only need the fix of  the `rust-url` parsing issue in `servo`.

**Pull request merged**